### PR TITLE
Make thumbnail link null for audio files without artwork

### LINKED ISSUE #528: Possibly make `thumbnail` null for audio files without artwork
## Description

Currently the frontend tries to fetch thumbnails for all audio files regardless of whether the audio file in question has one or not. 
I noticed that the API returns the thumbnail URL for all tracks. That makes sense, but could we improve this to be `null` for audio tracks without artwork? Then we could check the field in the frontend before making a network request.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #528 by @zackkrida.
Should also fix WordPress/openverse-frontend#1475

## Description

Adds a check to the `AudioSerializer` if the underlying audio row has artwork and removes the link in the case is not found. I believe we should make sure the cover of the album (if exists) is duplicated in each track if that track does not have its own artwork. In fact, I find it rare the case where the audio will have a different cover than their album.

This also seems to be the common case with our current live data, the artwork of the `audio_set` will never be required unless you are watching at the `audio_set` directly.

```sql
-- Audio with thumbnail & audio set
 SELECT COUNT(id) FROM audio
 WHERE thumbnail IS NOT NULL AND audio_set_foreign_identifier IS NOT NULL;
 
+-------+
| count |
|-------|
| 0     |
+-------+
SELECT 1
Time: 0.179s


-- Audio with thumbnail & without audio set
 SELECT COUNT(id) FROM audio
 WHERE thumbnail IS NOT NULL AND audio_set_foreign_identifier IS NULL;
 
+-------+
| count |
|-------|
| 89138 |
+-------+
SELECT 1
Time: 0.179s


-- Audio without thumbnail & with audio set
 SELECT COUNT(id) FROM audio
 WHERE thumbnail IS NULL AND audio_set_foreign_identifier IS NOT NULL;
 
+-------+
| count |
|-------|
| 0     |
+-------+
SELECT 1
Time: 0.177s
```

## Testing Instructions

Find an audio row without a thumbnail in the database and check if is null in the `audio_detail` response.
Try with this search for the current data sample:
```
localhost:8000/v1/audio?q=zaus&filter_dead=false
```

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — APPROVED** [review]
This is a great fix for the problem that's been annoying me for so long! 🎉
I wonder if we should the check to the indexing phase, though. It seems that this will add one extra db request per audio now, wouldn't it?

**Comment by krysal** [comment]
Yes, that's correct. I wanted to avoid it but seems related to Elasticsearch mapping, it doesn't return the thumbnail in the `Hit` instance so I had to look for the row in the database to solve the problem here (similar case to `peaks`), while we figure out the best way to add those non-search fields. We should reduce the number of times we look for a DB row in the future for sure.

**Comment by dhruvkb** [comment]
This seems quite suboptimal performance wise. IIRC this n+1 requests was the main reason we precomputed all the waveforms.

Can we push this till we make our ES indices improvements and add thumbnails to ES `_source`?

**Comment by krysal** [comment]
> This seems quite suboptimal performance wise. IIRC this n+1 requests was the main reason we precomputed all the waveforms.

I agree that is suboptimal. This is not intended to be a permanent solution but to fix the many requests we're making in the [frontend](https://github.com/WordPress/openverse-frontend/issues/1475) asking for non-existent audios' cover, as the ES indices changes got delayed by API outages. The serializer is still making a query per audio to add `peaks` so the heavy task is on the actual generation of the waveform.

> Can we push this till we make our ES indices improvements and add thumbnails to ES _source?

I'm afraid is taking too long but I'm really curious to explore the `_source` field. I hope to make time this week to explore an alternative solution.

File: api/catalog/api/serializers/audio_serializers.py [hunk 1]
```diff
@@ -135,6 +135,17 @@ def get_peaks(obj) -> list[int]:
             obj = Audio.objects.get(identifier=obj.identifier)
         return obj.get_waveform()
 
+    def to_representation(self, instance):
+        # Get the original representation
+        output = super().to_representation(instance)
+
+        if isinstance(instance, Hit):
+            audio = Audio.objects.get(identifier=instance.identifier)
```
Comment (AetherUnbound) [inline api/catalog/api/serializers/audio_serializers.py:143]: Given that we know this isn't a permanent solution, do you mind adding a `# TODO: Remove when <condition>` comment here so we can ensure this is removed when we make the ES changes?

**Review by AetherUnbound — APPROVED** [review]
Looks good, just one request! 🚀

**Pull request merged**